src/lib/hooks/useCheckApprovals.tsx

```diff
@@ -0,0 +1,45 @@
+import { useZnsContracts } from 'lib/contracts';
+import { tryFunction } from 'lib/utils';
+import React from 'react';
+
+interface CheckApproveHook {
+	tokenOperator: string | null;
+	approvedForAllTokens: boolean | null;
+}
+
+export function useApprovals(
```

Comment:
This was a good way to solve this if we were to use the data from this in a UI component directly, but we most likely will be using these methods in some business logic on-demand. As such, can you please refactor these to be functions, instead of state variables. If we need caching, we can implement that at another layer.

Also please include these methods in the existing `useApprovals.tsx` hook that you made instead of a separate hook like you did here.